cmd/agent/main.go

```diff
@@ -65,7 +64,9 @@ func run(cfg Config) {
 				log.Error(err)
 			}
```

Comment:
```suggestion
			} else {
				pollCounter = 0
			}
```
Если PollCounter отправить не удалось, лучше его не сбрасывать, иначе на сервере будет неактуальное значение. Не ставлю это, как блокер, но надеюсь, будет исправлено в следующих итерациях.